src/main/java/edu/harvard/iq/dataverse/MetadataBlock.java

```diff
@@ -101,17 +101,18 @@ public void setDatasetFieldTypes(List<DatasetFieldType> datasetFieldTypes) {
     }
     
     public boolean isDisplayOnCreate() {
-        // relying on "should" doesn't seem to work in context of a template 
-        // adding a transient that is updated in the DatasetVersionUI to fix
+        //Localize case - e.g. being called in the context of a specific collection 
+        if (getLocalDisplayOnCreate() != null){
+            return getLocalDisplayOnCreate();
+        }
+        // Non-localized case - the datasetFieldTypes are straight from the database and
+        // never have dsfType.localDsiplayOnCreate set.
         for (DatasetFieldType dsfType : datasetFieldTypes) {
-            boolean shouldDisplayOnCreate = dsfType.shouldDisplayOnCreate();
+            boolean shouldDisplayOnCreate = dsfType.isDisplayOnCreate();
```

Comment:
This change is just for clarity - since dsfType.localDisplayOnCreate can't ne non-null here, shouldDisplayOnCreate and isDisplayOnCreate are always the same.